.github/workflows/deploy.yml

```diff
@@ -190,15 +211,19 @@ jobs:
         id: restore-build
         with:
           path: |
+            ./packages/get-starknet/dist/webpack
             ./packages/wallet-ui/build
+            ./packages/starknet-snap/package.json
+            ./packages/starknet-snap/dist
+            ./packages/starknet-snap/snap.manifest.json
             ./node_modules/.yarn-state.yml
           key: ${{ github.sha }}
       - name: Deploy to AWS
         run: | 
           echo "Deployed Dapp to : $AWS_S3_URL"
           aws s3 sync ./packages/wallet-ui/build "$AWS_S3_URL"
         env:
-          AWS_S3_URL: ${{ vars.AWS_S3_URL }}
+          AWS_S3_URL: ${{ needs.perpare-deployment.outputs.AWS_S3_URL }}
```

Comment:
why do we use needs.perpare-deployment.outputs instead of vars here ? Also there might be a typo on needs.perpare maybe it should be needs.prepare ?

Reply:
yup , i will fix the typo: needs.prepare

and yes , using vars u have to take an extra AGR on the workflow, telling the workflow to extra param from action parameter store

hence, i use the prepare parameter workflow to do everything, and output the required var

and the following workflow will just need to use needs.perpare-deployment to get the param